doc/nrf/protocols/matter/getting_started/matter_api.rst

```diff
@@ -0,0 +1,248 @@
+The value passed into this callback is the value to which the attribute was set by the framework.
+Apart from the value, this function is called with the attribute path ( of ``chip::app::ConcreteAttributePath``type) that can be used to filter the cluster and particular attribute of interest.
+:c:func:`MatterPostAttributeChangeCallback()` function is useful in case there is a need to provide the synchronization between the Data Model and the application state.
+For instance, the Matter device implementing a light bulb may drive the state of LED based on the ``On/Off`` attribute value.
+Every change of this attribute is reported by aforementioned callback and thus be captured in the application layer.
+Apart from :c:func:`MatterPostAttributeChangeCallback()`, Matter defines other generic callbacks that can be employed in different use cases.
+As an example, :c:func:`emberAfExternalAttributeReadCallback()` and :c:func:`emberAfExternalAttributeWriteCallback()` can be used to store and handle attributes externally, by bypassing the Matter Data Model framework.
+To learn the complete set of Matter generic callbacks please use the :file:`ncs/modules/lib/matter/src/app/util/generic-callbacks.h` header file and included Doxygen commentary a reference.
+The exemplary implementation of the :c:func:`MatterPostAttributeChangeCallback()` that can be used to control the DoorLock Matter device type is listed below:
+
+.. code-block:: C++
+
+   #include <app-common/zap-generated/ids/Clusters.h>
+   #include <app/ConcreteAttributePath.h>
+   #include <app/util/generic-callbacks.h>
+
+   using namespace ::chip;
+   using namespace ::chip::app::Clusters;
+
+   void MatterPostAttributeChangeCallback(const chip::app::ConcreteAttributePath &attributePath, uint8_t type,
+                   uint16_t size, uint8_t *value)
+   {
+     if (attributePath.mClusterId == DoorLock::Id &&
+         attributePath.mAttributeId == DoorLock::Attributes::LockState::Id) {
+       /* Post events only if current lock state is different than given */
+       switch (*value) {
+       case to_underlying(DlLockState::kLocked):
+         /* Lock the physical door lock. */
+         break;
+       case to_underlying(DlLockState::kUnlocked):
+       /* Unlock the physical door lock. */
+         break;
+       default:
+         break;
+       }
+     }
+   }
+
+Apart from Matter generic callbacks, Matter Data Model engine provides callbacks that are cluster specific.
+These callbacks are usually defined was weak functions in the :file:`callback-stub.cpp` file that is generated together with other C++ source files when configuring clusters automatically with the `ZCL Advanced Platform`_ (ZAP tool).
+In case of some clusters however, the different approach is used and related callbacks are defined within the source code that constitutes the implementation of the cluster itself.
```

Comment:
```suggestion
In case of some clusters, however, a different approach is used and related callbacks are defined within the source code that constitutes the implementation of the cluster itself.
```